src/main/java/com/projecturanus/betterp2p/client/gui/GuiAdvancedMemoryCard.kt

```diff
@@ -1,54 +1,68 @@
 package com.projecturanus.betterp2p.client.gui
 
 import appeng.client.gui.widgets.MEGuiTextField
-import com.projecturanus.betterp2p.BetterP2P
 import com.projecturanus.betterp2p.MODID
 import com.projecturanus.betterp2p.capability.MemoryInfo
-import com.projecturanus.betterp2p.capability.P2PTunnelInfo
 import com.projecturanus.betterp2p.client.ClientCache
 import com.projecturanus.betterp2p.client.TextureBound
-import com.projecturanus.betterp2p.client.gui.widget.IGuiTextField
-import com.projecturanus.betterp2p.client.gui.widget.WidgetP2PDevice
-import com.projecturanus.betterp2p.client.gui.widget.WidgetScrollBar
-import com.projecturanus.betterp2p.item.BetterMemoryCardModes
+import com.projecturanus.betterp2p.client.gui.widget.*
 import com.projecturanus.betterp2p.network.*
-import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.FontRenderer
 import net.minecraft.client.gui.GuiButton
 import net.minecraft.client.gui.GuiScreen
 import net.minecraft.client.renderer.Tessellator
 import net.minecraft.client.resources.I18n
 import net.minecraft.util.ResourceLocation
-import net.minecraftforge.common.util.ForgeDirection
 import org.lwjgl.input.Keyboard
 import org.lwjgl.input.Mouse
+import org.lwjgl.opengl.GL11
 import java.util.*
 
+const val GUI_WIDTH = 288
+const val GUI_TEX_HEIGHT = 264
 class GuiAdvancedMemoryCard(msg: S2CListP2P) : GuiScreen(), TextureBound {
 
-    private val xSize = 288
-    private val ySize = 242
-    private val guiLeft: Int by lazy { (width - this.xSize) / 2 }
-    private val guiTop: Int by lazy { (height - this.ySize) / 2 }
+    private lateinit var _guiLeft: Lazy<Int>
+    private var guiLeft: Int
+        get() = _guiLeft.value
+        set(value) {
+            _guiLeft = lazyOf(value)
+        }
+
+    private lateinit var _guiTop: Lazy<Int>
+    private var guiTop: Int
```

Comment:
why are these necessary and you don't just use `this.guiTop` from the superclass?

Reply:
Those fields don't exist.

Reply:
Ah right those are in GuiContainer, not GuiScreen